dbms/src/Common/UTF8Helpers.h

```diff
@@ -49,9 +56,47 @@ inline size_t seqLength(const UInt8 first_octet)
 inline size_t countCodePoints(const UInt8 * data, size_t size)
 {
     size_t res = 0;
+    const auto end = data + size;
+
+#if __SSE2__ && __SSE4_1__
+    const auto bytes_sse = sizeof(__m128i);
+    const auto src_end_sse = (data + size) - (size % bytes_sse);
+
+    const auto one_sse = _mm_set1_epi8(1);
+    const auto zero_sse = _mm_setzero_si128();
+
+    const auto align_sse = _mm_set1_epi8(0x40);
+    const auto upper_bound = _mm_set1_epi8(0xBF);
+
+    for (; data < src_end_sse;)
+    {
+        auto sse_res = _mm_set1_epi8(0);
+
+        for (int i = 0, limit = 0xFF; i < limit && data < src_end_sse; ++i, data += bytes_sse)
+        {
+            const auto chars = _mm_loadu_si128(reinterpret_cast<const __m128i *>(data));
+
+            ///Align to zero for the solve two case
+            const auto align_res = _mm_adds_epu8(chars, align_sse);
+            const auto less_than_and_equals = _mm_cmpeq_epi8(_mm_min_epu8(align_res, upper_bound), align_res);
+            ///Because _mm_cmpeq_epi8 return 0xFF if true
+            const auto choose_res = _mm_blendv_epi8(zero_sse, one_sse, less_than_and_equals);
+
+            sse_res = _mm_adds_epu8(sse_res, choose_res);
```

Comment:
Probably it is better to use typical approach with movmask and popcnt.
It looks like this:

```
UInt16 bit_mask = _mm_movemask_epi8(byte_mask);
count += __builtin_popcount(bit_mask);
```

but you should test.

Reply:
Yes, it's a better choice. I've changed it.